### ISSUE TITLE
Use of `<ref>`

### ISSUE THREAD

**Issue by michaelnmmeyer** [erc-dharma/project-documentation#298]
A minor remark I am not sure I made: when referring to inscriptions with `<ref>` (EGD §10.4.6.Referring to inscriptions in the DHARMABase), the use of `@n` for indicating a repository is not needed (because all texts share a single namespace), and adding the `.xml` extension is also unnecessary.

Thus

`<ref n="tfa-pallava-epigraphy" target="INSPallava00001.xml">Pallava 1</ref>`

can be written

`<ref target="INSPallava00001">Pallava 1</ref>`

Internally, all variations `DHARMA_INSPallava00001`, `DHARMA_INSPallava00001.xml`, and `INSPallava00001`, `INSPallava00001.xml` are made to point to https://dharmalekha.info/texts/INSPallava00001.

**Comment by danbalogh** [comment]
Thanks, I have not heard this before. I'm very happy about lifting the part about using n to identify the repository. As for the filenames, I think it would be better to keep them systematic and use the full prefix all the time, perhaps also the .xml extension. That way there is, I think, a better chance of interpreting the files if someone else comes across them years from now and does not necessarily have access to your processing. If you disagree with that and say that it is technically not a problem, then I accept that, but even then, I would prefer to have just a single "legitimate" way to encode such references. I don't think it's good to have a guide that says "you can do A, B, C or D and it doesn't matter which, so just pick whichever you like at the moment".
So if you are sure that using the full filename (`DHARMA_INSPallava00001.xml`) is not better practice or more more future-proof in any way than the others, then let's pick the simplest form (`INSPallava00001`) as the only approved solution; otherwise, let's use the full filename and make the schema flag anything else as an error instead of processing it to make it work.

Whichever is chosen, the schema may need some alterations. At the moment, @n still comes up in Oxygen as a permitted attribute for `<ref>`, and I think there are no other circumstances in which one would want an n on a ref. So that can be discarded, and the existing instances of n on ref can be deleted from the files. For ref, I also get a suggestion list consisting of the filenames in the same folder as the present file, which I like, since most of my crossreferences are to my own subcorpus. But if we enforce a short reference instead of the full filename, then this needs to be changed.

**Comment by michaelnmmeyer** [comment]
Using the full file name is indeed probably the safest bet, if only for the autocomplete feature you pointed out. This might also discourage people from inventing file names; I am seeing a lot of names like `AirAsih.xml`, `WanuaTengahIII`, etc., that cannot be resolved to a real file.

**Comment by danbalogh** [comment]
Right. Shall we then agree that `<ref target="DHARMA_INSPallava00001.xml">Pallava 1</ref>` will be the only acceptable form of encoding a reference to a DHARMA inscription edition?

**Comment by arlogriffiths** [comment]
Curiously, I had never been aware of the tehoretical obligatuon to use @n on `<ref>` and in blissful unawareness been encoding as per the result of your discussion. I have no objections.

@michaelnmmeyer : can you extract a list of non-compliant names like AirAsih.xml and WanuaTengahIII.xml, so I can fix them? I hope you are aware that in general all file names bearing the string IDENK are not yet FNC-compliant in that they use inscription names rather than numbers, as temporary solution while we are waiting for the IDENK database (idenk.net) to deliver inscription numbers for these items. This will start to happen within the next half year, I hope.

**Comment by michaelnmmeyer** [comment]
@arlogriffiths 

Here is the list of references. It is probably too long to be useful, though.

> ...
> Adan-adan.xml
> Adanadan.xml
> Adan-Adan.xml
> AirAsih.xml
> BadamiCalukya00004-Kopparam-Pulakesin.xml
> Balawi.xml
> Baru.xml
> BengalCharters00066.xml
> Bimalasrama.xml
> C00017.xml
> C0030A1.xml
> C0030A3.xml
> C0030A.xml
> C0030B4.xml
> C0030B.xml
> C0081.xml
> C0087.xml
> C0096.xml
> Cane.xml
> Canggu.xml
> cic-bibl:goodall-2004
> Dk0019.xml
> Dk0020.xml
> EIAD0018.xml
> EIAD0105.xml
> EIAD0161.xml
> EIAD0168.xml
> eiad-bibl:Ramachandranb.xml
> eiad-bibl:Subba_Rao1928.xml
> Gandhakuti
> Gandhakuti.xml
> Garung.xml
> Guntur.xml
> GunungWukir.xml
> Haliwangbang.xml
> #inv-general
> K0011.xml
> K0056.xml
> K0077.xml
> K0136.xml
> K0175.xml
> K0379.xml
> K0417.xml
> K0444.xml
> K0468.xml
> K0524.xml
> K0582.xml
> K0682-1.xml
> K0856.xml
> K0868.xml
> K1281.xml
> K1282.xml
> K1284.xml
> kakurugan.xml
> Kakurugan.xml
> Kalimusan.xml
> Kamalagyan.xml
> KambangPutih.xml
> Kancana.xml
> Kawali_2.xml
> Kurungan.xml
> kusambyan.xml
> Laguna.xml
> Landa.xml
> Luitan.xml
> Mādhavapura.xml
> Madhawapura.xml
> manahimanuk.xml
> Marsmu.xml
> Mula-Malurung.xml
> Munggut.xml
> PakisWetan.xml
> Palebuhan.xml
> Palepangan.xml
> Parablyan.xml
> Paralabyan.xml
> Parhyangan.xml
> Parung_II.xml
> Patitihan.xml
> Plumbangan.xml
> Rukam.xml
> Rumwiga.xml
> Sanga.xml
> Sarvadharma.xml
> Selamandi_I.xml
> simaanglayang.xml
> Sindoro-PuBahut
> Sindoro-PuBahut.xml
> Sindoro-PuMamruk.xml
> Sobhamerta.xml
> Śobhāmṛta.xml
> Sukhamerta.xml
> Taragal.xml
> tfbBengalCharters00067.xml
> tfbBengalCharters00068.xml
> tfbBengalCharters00069.xml
> TigaRon.xml
> Tuhañaru.xml
> tuhanyaru.xml
> TurunHyang.xml
> Vukiran.xml
> WaharuIV.xml
> WanuaTengahIII.xml
> WaringinPitu.xml
> WintangMasB.xml

**Comment by danbalogh** [comment]
BadamiCalukya00004-Kopparam-Pulakesin.xml was in my corpus, now corrected.

**Comment by arlogriffiths** [comment]
Thanks. I have converted the above list to a task list with check boxes and started weeding out offending cases listed above.
@ryosukefurui @ekobastiawan @tyassanti @chhomkunthea @salomepichon @chloechollet @wayanjarrah : Please read the above discussion. Then please help make our files compliant to the precise rules for use of `<ref>`. Search cases of offending strings using the "Find/Replace in Files" function, choosing the appropriate repository where the cases is suspected to occur. Correct the relevant file and check the item in the list above. 

Examples of correct references for the tfc repositories:

`<ref target="DHARMA_INSCIK00011.xml">K. 11</ref>`
`<ref target="DHARMA_INSCIC00017.xml">C. 17</ref>`
`<ref target="DHARMA_INSIDENKWintangMasB.xml">Wintang Mas B</ref>`

Don't hesitate to ask if anything more needs to be explained.

<img width="652" alt="Capture d’écran 2024-05-06 à 11 51 23" src="https://github.com/erc-dharma/project-documentation/assets/835603/dff87f2f-4c1f-4cc7-a6b3-842294ded325">

**Comment by ekobastiawan** [comment]
@michaelnmmeyer : Do you know why I can't tick the boxes above?

**Comment by michaelnmmeyer** [comment]
@ekobastiawan I have no idea. This might require an administratror account.

**Comment by arlogriffiths** [comment]
I have added @ekobastiawan among assignees. Can you try again now, Eko? 

If that too fails, we will need to split up the above list and create separate list per repo. But even on tfd-nusantara-epigraphy, does Eko have administrator rights?

**Comment by danbalogh** [comment]
I am able to tick and untick. I'm assigned, but not an admin as far as I know. So I guess Eko should fine now that he is assigned.

**Comment by ekobastiawan** [comment]
@arlogriffiths : I still can't tick the boxes

**Comment by arlogriffiths** [comment]
I suspect the problem has to do with other-than-admin-level access to the repo, which Dan does but Eko doesn't have.

Can you look into this, @michaelnmmeyer? Can we do something about it?

**Comment by danbalogh** [comment]
Sounds logical. I'm afraid I have no idea how to check my level of access.

**Comment by manufrancis** [comment]
@ekobastiawan 
Make sure you have sign in to your github account

**Comment by ekobastiawan** [comment]
@michaelnmmeyer Thanks a lot, I am now able to tick the boxes

**Comment by chhomkunthea** [comment]
Dear all,

I cannot check the boxes in the list above. I think that I am already signed in the Github. Maybe I have not been given access or was at the wrong place. Can you please help?  

Best,
Kunthea

**Comment by salomepichon** [comment]
Dear all,

I've for now made the modifications for the cam corpus. I haven't been able to locate the cases of C0087.xml
and C0096.xml, however.

**Comment by arlogriffiths** [comment]
@michaelnmmeyer : 

1. can you give @chhomkunthea write access in the same way that you did for Eko?
2. like @salomepichon, I am unable to find cases of C0087.xml and C0096.xml in the tfc-campa-epigraphy repo. Where should we be looking?

**Comment by chhomkunthea** [comment]
Dear all,

There is also problem in my files (K. 11, K. 56, K. 77, K. 417 and K. 582). Among them, only K. 56 has a <ref target="DHARMA ...> markup. And I don't see the K. 136.xml in the folder "xml-provisional".

Actually, there are files, especially the hospital inscriptions of Jayavarman VII (K. 12, K. 368, K. 375 ...) which contain many <ref> markups. They conform to the norm, i.e. without the @n. 

Best,
Kunthea

**Comment by arlogriffiths** [comment]
I think you may have misunderstood the nature of the list above. It is not a list of files to be opened and checked, but a list of strings to be searched (in your case in tfc-khmer-epigraphy) and to be replaced by the correct string. For example, if you use "Search/Replace in files" to search the string K0379.xml, you will find one occurrence, namely in the file DHARMA_INSCIK00216-S.xml. In that file, you need to replace `<ref target="K0379.xml">K. 379</ref>` by `<ref target="DHARMA_INSCIK0379.xml">K. 379</ref>` and then tick K0379.xml in the list above. Is it clear now?

**Comment by chhomkunthea** [comment]
Yes, it is. Thank you! But it seems that there is one zero missing in the file name, i.e. <ref target="DHARMA_INSCIK00379.xml">K. 379</ref>.

**Comment by chhomkunthea** [comment]
Yes, it is. Thank you! But it seems that there is one zero missing in the file name, i.e. 

<ref target="DHARMA_INSCIK00379.xml">K. 379</ref>.

**Comment by chhomkunthea** [comment]
Yes, it is. Thank you!

It seems that there is one zero missing in the file name. Should it be "DHARMA_INSCIK00379" instead of "DHARMA_INSCIK0379" ?

**Comment by arlogriffiths** [comment]
Indeed, small typo from my side. Sorry. Please do add that zero.

**Comment by chhomkunthea** [comment]
Well noted with thanks.

**Comment by chhomkunthea** [comment]
Dear all,

FYI, I have corrected the <ref> related to K0011 through K1284 in the list above. I hope that they are all fine now.

Best,
Kunthea

**Comment by michaelnmmeyer** [comment]
@chhomkunthea You should now be able to tick boxes.

**Comment by chhomkunthea** [comment]
Thank you very much! Yes, it's done now.

**Comment by michaelnmmeyer** [comment]
@arlogriffiths They have been corrected in the meantime.

**Comment by arlogriffiths** [comment]
Thanks.

@michaelnmmeyer : can you tell me where to look for Dk0019.xml and Dk0020.xml?

@ryosukefurui : all remaining items concern tfc-bengalcharters-epipgraphy: can you take care of them?

**Comment by ryosukefurui** [comment]
I have just corrected relevant ref in DHARMA_INSBengalCharters00065.xml, and ticked the list. Excuse me for a delayed response.

**Comment by michaelnmmeyer** [comment]
@arlogriffiths 

Dk0019.xml and Dk0020.xml are both in tfb-daksinakosala-epigraphy/texts/DHARMA_INSDaksinaKosala00021.xml

**Comment by arlogriffiths** [comment]
Thanks. So the remaining work for @NatasjaSB. I don't know if she is still following github, and anyhow I assume @danbalogh can easily make the small modifications in her xml files on her behalf.

So @danbalogh, could you take care of this and then close this issue?

**Comment by danbalogh** [comment]
I've made the correction in the DaksinaKosala file. Natasja has recently renamed her files at our request, to follow the pattern used in other collections (INSDaksinaKosala instead of INSDk), and I assume that she did not think to check for and update existing references to files when she did that rename. Her repository seems to contain no other obsolete references.
